src/Chapter10.Tests/Listing10.17.RegisteringAFinalizerWithProcessExit.Tests.cs

```diff
@@ -19,25 +21,32 @@ public class DisposeTests
 
     private const string ProjectName = "ProcessExitTestProgram.testing";
 
+    private readonly static object RunPowerShellScriptSyncLock = new();
     [ClassInitialize]
-    public static void ClassInitialize(TestContext _)
+    public static void ClassInitialize(TestContext testContext)
     {
+        string psOutput;
         // Clean up at the start in case the class cleanup doesn't run (due to debug for example)
-        Assert.AreEqual<int>(0, RunPowerShellScript("cleanup", out string _));
-        string testStage = "create";
-        Assert.AreEqual<int>(0, RunPowerShellScript(testStage, out string psOutput),
+        string testStage = "cleanup";
+        Assert.AreEqual<int>(0, RunPowerShellScript(testStage, out psOutput),
+            $"Script failed with $testStage='{testStage}'. psOutput:\n{psOutput}");
+        testStage = "create";
+        int exitCode = RunPowerShellScript(testStage, out psOutput);
```

Comment:
Do these executions also need to acquire the mutex?

Reply:
They are in a class initializer so should only run once and before anything else in the test. In summary, I don't think synchronization is required.

Reply:
That is not quite accurate. We are probably fine for now because we are just taking the defaults for MSTest, however it supports method level parallelization. Each execution of a test method will create a new instance of the test class, and if two tests are executed in parallel it is possible to have two threads executing this method at the same time.

https://learn.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022#mstest-element